src/display/text_layer.js

```diff
@@ -527,21 +537,57 @@ var renderTextLayer = (function renderTextLayerClosure() {
       this._capability.reject('canceled');
     },
 
+    _processItems(items, styleCache) {
+      for (let i = 0, len = items.length; i < len; i++) {
+        this._textContentItemsStr.push(items[i].str);
+        appendText(this, items[i], styleCache);
+      }
+    },
+
     _render: function TextLayer_render(timeout) {
-      var textItems = this._textContent.items;
-      var textStyles = this._textContent.styles;
-      for (var i = 0, len = textItems.length; i < len; i++) {
-        appendText(this, textItems[i], textStyles);
+      let capability = createPromiseCapability();
+      let styleCache = Object.create(null);
+
+      if (this._textContent) {
+        let textItems = this._textContent.items;
+        let textStyles = this._textContent.styles;
+        this._processItems(textItems, textStyles);
+        capability.resolve();
+      } else if (this._textContentStream) {
+        let pump = () => {
+          this._reader.read().then(({ value, done, }) => {
+            if (done) {
+              capability.resolve();
+              return;
+            }
+
+            Util.extendObj(styleCache, value.styles);
+            this._processItems(value.items, styleCache);
+            pump();
+
+          }, capability.reject);
+        };
+
+        this._reader = this._textContentStream.getReader();
+        pump();
+      } else {
+        throw new Error('Neither "textContent" nor "textContentStream"' +
+          ' parameters specified.');
       }
 
-      if (!timeout) { // Render right away
-        render(this);
-      } else { // Schedule
-        this._renderTimer = setTimeout(() => {
+      capability.promise.then(() => {
+        styleCache = null;
+        if (!timeout) { // Render right away
           render(this);
-          this._renderTimer = null;
-        }, timeout);
-      }
+        } else { // Schedule
+          this._renderTimer = setTimeout(() => {
+            render(this);
+            this._renderTimer = null;
+          }, timeout);
+        }
+      }, () => {
```

Comment:
>Replacing this line with `}, this._capability.reject);` instead ought to work just fine

Yes, you are right. This is working just fine.

But i am trying something like `}, this._capability.reject('Error while processing chunks'));`